### PR TITLE
MANTA-4821 Make buckets-api use the storinfo service

### PR DESCRIPTION
**Pull request by rejohnst**
I deployed these changes in my lab rig along with an instance of the Storinfo and verified that I could create buckets and copy objects into those buckets.

Khang also deployed these changes on his test rig and was able to successfully run cosbench against these changes.

Before I push these changes there will need to be a push to sdc-manta to ensure that the Storinfo service is deployed by default. (Today it has to be deployed manually)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
         "restify-clients": "2.6.7",
         "restify-errors": "^5.0.0",
         "sshpk": "1.16.1",
+        "storinfo": "git+https://github.com/joyent/node-storinfo.git",
```
Comment (bahamas10) [inline package.json]: Whether `node-storinfo` gets put on npm is a different discussion, but I think at the very least `node-storinfo` should push a `git tag` and then this `package.json` should use that tag/version.

Comment (rejohnst) [reply package.json]: Done - I tagged the commit in node-storinfo and then updated the package.json here to point to it.

**Review by kellymclaughlin — APPROVED** [review]
This looks good to me! I'm excited to see this finally come to fruition. Is there a PR for the sdc-manta changes already? Let me know if you need in help with that work.

**Comment by kellymclaughlin** [comment]
The CI check failure should go away if you merge in the latest changes from `master`.

**Comment by rejohnst** [comment]
> The CI check failure should go away if you merge in the latest changes from `master`.

Thanks!  I was just trying to figure what was happening there.

**Comment by rejohnst** [comment]
> This looks good to me! I'm excited to see this finally come to fruition. Is there a PR for the sdc-manta changes already? Let me know if you need in help with that work.

Not yet.  I'm working on that today.

**Comment by rejohnst** [comment]
> > This looks good to me! I'm excited to see this finally come to fruition. Is there a PR for the sdc-manta changes already? Let me know if you need in help with that work.
> 
> Not yet. I'm working on that today.

Actually - it looks like maybe Trent's changes for MANTA-4957 may have taken care of this already.

**Review by bahamas10 — APPROVED** [review]
:+1:

**Comment by rejohnst** [comment]
> > > This looks good to me! I'm excited to see this finally come to fruition. Is there a PR for the sdc-manta changes already? Let me know if you need in help with that work.
> > 
> > 
> > Not yet. I'm working on that today.
> 
> Actually - it looks like maybe Trent's changes for MANTA-4957 may have taken care of this already.

Trent's changes for MANTA-4957 take care of making sure storinfo is part of the default coal and lab installs.  AFAIK, We don't generate the manta config json from a script for production installs so there is no  code in sdc-manta to update for that.  However, the deployment documentation in the Manta Operator's Guide does need to be updated so that admins know that they need to include at least one Storinfo instance.  (The operators guide actually needs a lot of updates).  I've filed MANTA-5079 to track these changes.

Unless there are objections, I plan to merge this PR today.  Then I'll send a heads-up to ~manta describing the impact.

**Pull request merged**